tasks/main.yml

```diff
@@ -7,21 +7,41 @@
     absent_objects: "{{ objects | selectattr('state', 'defined') | selectattr('state', 'equalto', 'absent') | list |
                         map(attribute='object')| list }}"
 
+- name: Split objects into to be created and to be applied
+  set_fact:
+    present_create_objects: "{{ present_objects | selectattr('kind', 'equalto', 'ConfigMap') | list }}"
+    present_apply_objects: "{{ present_objects | rejectattr('kind', 'equalto', 'ConfigMap') | list }}"
+
+- name: Log objects to create
+  debug:
+    msg: "Creating {{ item.metadata.name }}"
+  loop: "{{ present_create_objects }}"
+  loop_control:
+    label: "{{ item.metadata.name }}"
+
+- name: Create resources
+  k8s:
+    resource_definition: "{{ present_create_objects }}"
+    validate:
+      fail_on_error: true
+      strict: true
+  when: present_create_objects | length > 0
```

Comment:
moet hier niet nog een extra stap
een check of het object niet al bestaat
want ik denk dat als het een object probeerd te createn dat al bestaat dat dat een error zou moeten geven
alternatief kan force:true worden aangezet, maar dat is ook een risisco

Reply:
Ik heb dit getest, maar als een object al bestaat wordt deze niet opnieuw aangemaakt. Dit zit in de kubernetes.core dependency ingebakken:
https://github.com/ansible-collections/kubernetes.core/blob/bf5d70e9204bfb0945d8b1d13dc875e2790da2fa/plugins/module_utils/common.py#L647